core/src/main/java/com/klaytn/caver/transaction/type/ChainDataAnchoring.java

```diff
@@ -0,0 +1,214 @@
+        rlpTypeList.add(RlpString.create(Numeric.toBigInt(this.getNonce())));
+        rlpTypeList.add(RlpString.create(Numeric.toBigInt(this.getGasPrice())));
+        rlpTypeList.add(RlpString.create(Numeric.toBigInt(this.getGas())));
+
+        rlpTypeList.add(RlpString.create(Numeric.hexStringToByteArray(this.getFrom())));
+        rlpTypeList.add(RlpString.create(Numeric.hexStringToByteArray(this.getInput())));
+        rlpTypeList.add(new RlpList(signatureRLPList));
+
+        byte[] encodedTransaction = RlpEncoder.encode(new RlpList(rlpTypeList));
+        byte[] type = new byte[] { (byte)TransactionType.TxTypeChainDataAnchoring.getType() };
+        byte[] rawTx = BytesUtils.concat(type, encodedTransaction);
+
+        return Numeric.toHexString(rawTx);
+    }
+
+    /**
+     * Returns the RLP-encoded string to make the signature of this transaction.
+     * @return String
+     */
+    @Override
+    public String getCommonRLPEncodingForSignature() {
+        // SigRLP = encode([encode([type, nonce, gasPrice, gas, from, anchoredData]), chainid, 0, 0])
+        // encode([type, nonce, gasPrice, gas, from, anchoredData])
+
+        byte type = (byte)TransactionType.TxTypeChainDataAnchoring.getType();
+
+        List<RlpType> rlpTypeList = new ArrayList<>();
+        rlpTypeList.add(RlpString.create(type));
+        rlpTypeList.add(RlpString.create(Numeric.toBigInt(this.getNonce())));
+        rlpTypeList.add(RlpString.create(Numeric.toBigInt(this.getGasPrice())));
+        rlpTypeList.add(RlpString.create(Numeric.toBigInt(this.getGas())));
+        rlpTypeList.add(RlpString.create(Numeric.hexStringToByteArray(this.getFrom())));
+        rlpTypeList.add(RlpString.create(Numeric.hexStringToByteArray(this.getInput())));
+
+        byte[] encoded = RlpEncoder.encode(new RlpList(rlpTypeList));
+        String encodedStr = Numeric.toHexString(encoded);
+
+        return encodedStr;
+    }
+
+    /**
+     * Check equals txObj passed parameter and Current instance.
+     * @param obj The AbstractTransaction Object to compare
+     * @param checkSig Check whether signatures field is equal.
+     * @return boolean
+     */
+    @Override
+    public boolean compareTxField(AbstractTransaction obj, boolean checkSig) {
+        if(!super.compareTxField(obj, checkSig)) return false;
+        if(!(obj instanceof ChainDataAnchoring)) return false;
+        ChainDataAnchoring txObj = (ChainDataAnchoring)obj;
+
+        if(!this.getInput().equals(txObj.getInput())) return false;
+
+        return true;
+    }
+
+    /**
+     * Getter function for input
+     * @return String
+     */
+    public String getInput() {
+        return input;
+    }
+
+    /**
+     * Setter function for input
+     * @param input Data of the service chain.
+     */
+    public void setInput(String input) {
+        if(input == null) {
+            throw new IllegalArgumentException("input is missing.");
+        }
+
+        if(!input.equals("0x") && !Utils.isHex(input)) {
```

Comment:
Will Utils.isHex function return false with "0x"?

Reply:
Yes, it returns false.